### DISCUSSION TITLE
[Suggestion] Improvements to the infected gamemode

### DISCUSSION THREAD

**Discussion by Bobbyperson** [R2Northstar/Northstar#161] [Ideas]
## Main Suggestions:
The current infection gamemode is unbalanced and generally unfair towards infected players, I believe the following will make the gamemode more fair. I've been hosting and playing infection for a long time now, and I've made most of these changes, which are generally agreed upon by my community to be a drastic improvement from the original gamemode.

### The Survivor Meta
The meta is and always will be grapple with gravity star or smoke grenade. Why? Because with grapple, you can reach high places that are impossible to get to without it, go out of the map, and even fling to get to high speeds that stim can't reach. While the other tacticals can be good in some ways, especially if survivors stick together as a team, they will never be as good as grapple. On the default gamemode, it is literally impossible for the infected to win.

#### Solutions to this:
While you could just forbid this in your server rules, there will always be rule breakers or people who just don't know and you can't watch over 24/7. The easiest and fairest solution I've found is to give infected a grapple. This by itself is really unfair, so to balance it out further my changes are as follows:

1. Remove infected permanent stim effect
2. Give infected a grapple in the tactical slot
3. Give infected stim ability in the ordinance slot
4. Slightly reduce infected health

On paper this may seem broken or unfun, but I've found that rounds are a lot more even now. Survivors can't go to impossible to reach spaces anymore, and its no longer virtually impossible to run from infected. There have been a few games where the last survivor won through pure movement skill, even with 23 infected chasing them. I've also found its much more fun to be infected, as you have to use your abilities sparingly instead of getting insane movement speed and health regen for free.

### Infected Health Issue
The current equation for determining infected health is this: `(X+1)*10` where X is the amount of survivors alive. So if there are 10 survivors alive, each infected gets 110 health. I've found that this equation works best when the server is at 12-16 people, and gets a bit unfair when there is more than 16 or less than 12 total players. My server averages 24 people, so I've been tweaking it a bit lower, to 8 health per survivor alive. The main issue that arises with the default value and too many people is: when you have the right amount of infected, it becomes trivially easy to swarm a group of survivors. 

On a somewhat related note: This [graph](https://fox.girlsare.life/754dT7n.png) shows that the total infected health (added between all players) is at its peak when around half of the survivors have died. 

#### Possible solution:
When there are fewer players, infected should each have more health, and when there are more players, infected should have less health. This is to encourage swarming together in groups, instead of just tanking bullets one by one.
I think the equation should be edited to accommodate different player counts. While I'm not good enough at math to make an equation that works with most if not all cases, I can give a list of the amount of health I think the infected should get per survivor alive:

5-10 players: 12 health per survivor
10-16 players 8-10 health per survivor
17-24 players: 6-8 health per survivor
(Open to discussion on this)

### Abusable bugs/unintended features
- As a survivor you can escape infected by running out of bounds, as the out of bounds timer kills you, but doesn't switch your team to infected.
- You can also use the above bug to switch loadouts midmatch.
- As a survivor on the respawn screen, you can stay there indefinitely and guarantee yourself a win
- As a survivor if you throw any explosive/firestar/whatever and die by it, you and the throwable's team affiliation switches to infected, which can then kill other survivors. (People have been doing this on survivor spawn and when theres a group of survivors huddled together)
- If the timer is below 40s and the second to last survivor dies, the timer will be set to 40s regardless. (unsure if this is intentional or not)
- If you equip the Grand Theft Semi-Auto execution as an infected and spam E during the animation, you can pick up the survivor's gun.

These are all my suggestions for now, I'm very open to discussing all of this or hearing other people's suggestions.

**Comment by x3Karma** [comment]
Really great read. Have been considering adding a one charge refillable grapple for Infected, they are however unable to use it to hook survivors.

For the Infected health, currently the health stays static until you die. Ex: You spawn as Infected with 11 Survivors, so you have 120 health. Even though you continue killing, as long as you don't die as this Infected, your health will remain at 120 hp even if Survivor count is dwindling.

Proposed solution is to update all Infected's health when a Survivor dies.

The out of bound bug should correctly swap your team to Infected. The Survivor teamkilling bug is still present though (ngl its pretty funny)

**Comment by Bobbyperson** [comment]
I didn't even realize infected health didn't get updated, is it possible to change the max health without changing the current health on survivor death? As to avoid giving infected a free instant health regen.

On my server at least the out of bound bug doesn't switch your team even if you do it multiple times.

The survivor teamkilling bug is funny the first time but gets quickly frustrating, especially near the end of a round.

One more thing I forgot to mention, if the time limit is below 40s when there's one survivor left the time will be set to 40s regardless, I'm not sure if this is intentional or not.

**Comment by TrueCP6** [comment]
I am totally for the infected base movement speed reduction. As it is now it is trivially easy to kill survivors playing normally as it is impossible for them to outrun you, however as soon as survivors go out of bounds or to an unintended part of the map it becomes basically impossible to reach them as your vertical mobility as an infected is practically nonexistent.

**Comment by Bobbyperson** [comment]
After another month of hosting an infected server, I have come up with (in my mind) the perfect infected gamemode. To me, this will make it stand out more from infected in other games while still retaining the core gameplay. I'm not sure if all of this is possible, but I can dream. For clarification, this would be in conjunction with my original post.

### Main premise
The survivors spawn in spread out around the map in order to mimic a game in progress interrupted by the infection
The game goes on as it would in normal infection with some key differences:
- After a certain amount of infected kills, your faction leader will reward you with additional AI help like grunts or specters, and maybe the infected could get prowler npcs 
- At the end of the round, the remaining survivors have to escape to an evac ship in order to get a true win (optimally the infected don't know where the evac ship is going to land)
- The last survivor will still get a titan but still has to escape to the evac ship, which will come sooner

To me, I think this would be an immense improvement from the current gamemode. It encourages more teamwork and cooperation between survivors instead of hiding by yourself, while at the same time discourages camping in one single spot. This also would use Titanfall's unique mechanics to its advantage, making it stand out more from other games like Halo.

Please let me know what you guys think, I would love to hear your input.
